packages/cli-config-apple/src/tools/installPods.ts

```diff
@@ -10,27 +10,55 @@ import {
   runSudo,
 } from '@react-native-community/cli-tools';
 import runBundleInstall from './runBundleInstall';
+import path from 'path';
 
 interface PodInstallOptions {
   skipBundleInstall?: boolean;
   newArchEnabled?: boolean;
   iosFolderPath?: string;
+  platform: string;
+  root: string;
 }
 
 interface RunPodInstallOptions {
+  root: string;
+  platform: string;
   shouldHandleRepoUpdate?: boolean;
   newArchEnabled?: boolean;
 }
 
-async function runPodInstall(loader: Ora, options?: RunPodInstallOptions) {
+async function runPodInstall(loader: Ora, options: RunPodInstallOptions) {
   const shouldHandleRepoUpdate = options?.shouldHandleRepoUpdate || true;
   try {
+    if (fs.existsSync('build')) {
+      fs.rmSync('build', {recursive: true});
+    }
+
     loader.start(
       `Installing CocoaPods dependencies ${chalk.bold(
         options?.newArchEnabled ? 'with New Architecture' : '',
       )} ${chalk.dim('(this may take a few minutes)')}`,
     );
 
+    const reactNativePath = path.dirname(
+      require.resolve('react-native', {paths: [process.cwd()]}),
+    );
+    const codegenScript = path.join(
+      reactNativePath,
+      'scripts',
+      'generate-codegen-artifacts.js',
+    );
+
+    execa.sync('node', [
+      codegenScript,
+      '-p',
+      options.root,
+      '-o',
+      process.cwd(),
+      '-t',
+      options.platform,
+    ]);
+
```

Comment:
i would like to split it in a separate function and not couple it with the runPodInstall function. In this way, when we drop CocoaPods, you'll still have the `runCodegen` function in place and you can just remove the `runPodInstall` function